core/modules/catalog/helpers/index.ts

```diff
@@ -529,18 +530,66 @@ export function configureProductAsync (context, { product, configuration, select
  */
 
 export function getMediaGallery (product) {
-    let mediaGallery = []
-    if (product.media_gallery) {
-        for (let mediaItem of product.media_gallery) {
-            if (mediaItem.image) {
-                mediaGallery.push({
-                    'src': getThumbnailPath(mediaItem.image, rootStore.state.config.products.gallery.width, rootStore.state.config.products.gallery.height),
-                    'loading': getThumbnailPath(product.image, 310, 300)
-                })
-            }
-        }
+  let mediaGallery = []
+  if (product.media_gallery) {
+      for (let mediaItem of product.media_gallery) {
+          if (mediaItem.image) {
+              let computedImage = {
+                'src': getThumbnailPath(mediaItem.image, rootStore.state.config.products.gallery.width, rootStore.state.config.products.gallery.height),
+                'loading': getThumbnailPath(product.image, 310, 300),
+                'video': null
+              }
+              if (mediaItem.vid) {
+                computedImage.video = computeVideoData(mediaItem.vid)
+              }
+              mediaGallery.push(computedImage)
+          }
+      }
+  }
+  return mediaGallery
+}
+
+
+/**
+* Process video data to provide the proper 
+* provider and attributes
+* 
+* @param videoData Object
+*/
+
+function computeVideoData (videoData) {
+  let videoId = '';
+  let type = '';
+  let videoUrl = new Url(videoData.url);
+
+  if (videoUrl.host.match(/youtube\.com/) && videoUrl.search) {
+    videoId = videoUrl.search.split('v=')[1];
```

Comment:
parsing in this method looks a bit complicated. Maybe we'd use for example parseUrl from https://www.npmjs.com/package/query-string for that?
And then we could have this method during a data import to elastic?

Reply:
Regarding your suggestion to move this to the data import, I thought about that as well and it probably is the better solution. My only thought was if other platforms are integrated with the ability to import videos like this the url parsing logic would need to be recreated in each platform integration (e.g. mage2vuestorefront). I can move it to there unless you have a better suggestion as to where it could sit.

Reply:
The challenge with using `query-string` or `url-parser`, or any other module like this is that YouTube URLs have many different formats. When running some tests against different formats sometimes the `url-parser` I was using would get what I needed, and sometimes `query-string` would get it. But none of them work all the time. 
So, I cleaned it up and decided to use some Regex for both YouTube and Vimeo to match the domains and try to locate the video ID.